### ISSUE TITLE
default-authentication-plugin MySQL 8

### ISSUE THREAD

**Issue by laclaro** [NetKnights-GmbH/ubuntu#47]
It seems that the changed default-authentication-plugin in recent MySQL "requires a secure connection" when doing a connection for MASTER-SLAVE replication. The error looks like this

```
[ERROR] [MY-010584] [Repl] Slave I/O for channel '': error connecting to m
aster 'replicator@172.20.1.1:3306' - retry-time: 60 retries: 4 message: Authentication plugin 'caching_sh
a2_password' reported error: Authentication requires secure connection. Error_code: MY-002061
```

Our connection is not secured, but wrapped in a VPN tunnel usually, so the complaint is valid at this point.

The workaround is to set the default authentication plugin to native password _before_ creating the pi database user.

`default-authentication-plugin = mysql_native_password`

So this would be best to place in the default server installation package.

An alternative would be to change the configuration to use a secure connection on the transport-layer.

**Comment by plettich** [comment]
MySQL automatically generates a simple CA and SSL certificates during the installation, maybe we can utilize these for the secure connection (might need to exchange the CA certs beforehand).

**Comment by pablo-knight** [comment]
Has been implemented in pi-appliance #101
